docs/jsonapi-spec.md

```diff
@@ -401,6 +401,23 @@ The maximum size of field values are:
 The maximum length of an array is 100 elements.
 
 
+### Equality handling with arrays and subdocs
```

Comment:
The description says subdocs, but we dont have spec given for subdocs.

For subdocs it has to be exact match
`{'foo' : {"col1" : "bar1", "col2" :  "bar2"}}`, it has to match a field `foo` which has document value `{"col1" : "bar1", "col2" :  "bar2"}`

Reply:
Thanks Mahesh - will add clarification.

Reply:
Updated with latest commit.